### ISSUE TITLE
Allow createdat to be set/saved on new instances

### ISSUE THREAD

**Issue by chapmandu** [cfwheels/cfwheels#887]
// I am importing historical data
user.new(name="Jack Flash", createdat=oneYearAgo());
user.save();
// user.createdat is now equal to today

// workaround is to update
user.update(createdat=oneYearAgo());

**Comment by andybellenie** [comment]
Second this, it's always bugged me you can't manually set timestamps, same applies to updatedAt

**Comment by andybellenie** [comment]
You can avoid that issue by only doing an automatic updatedAt when the column value hasn't been changed, i.e. `if (!hasChanged('updatedAt')...`

**Comment by neokoenig** [comment]
My only concern in changing this is introducing a possibility of mass assignment with pre-existing apps; Let's assume you were doing this (which on a user object you'd not do, but you might do something like it somewhere) and let's assume the createdAt date was important to your application in some critical way:

```
// incoming form params: todo[title]: "foo"
todo = model("todo").create(params.todo)
```
The current behaviour has `createdAt` as an immutable value in this context and defaults to `now()`; with this change, if you get a spoofed form input which now has `createdAt` as some older date, you've got a value being altered which previously wouldn't have been. But then, thinking about it, this condition presumably already exists on `update()` too, so I'm probably overthinking it. Maybe the point is I always forget about timestamps

**Comment by andybellenie** [comment]
I agree in principle but I think the issue there is passing an unsanitized form input directly into the object. There are multiple ways to deal with that, but currently it's impossible to use automatic timestamps if you ever need to set it manually.

**Comment by chapmandu** [comment]
What's the consensus with this one?

**Comment by neokoenig** [comment]
Go ahead I think

**Comment by dbelanger** [comment]
I also think we should move ahead with this but I believe a flag is required to allow the overwriting of **any** auto-timestamp (deletedAt too for consistency).  This flag should default to false.

Similar to what we do with **allowSoftDeletes**, it would be **allowManualTimestampUpdates** or something to that effect.  This would protect against Tom's concern that it's a critical field in many existing apps .

**Comment by neokoenig** [comment]
I actually really like that idea - after all, when you're doing timestamp overrides, it's usually going to be something like "import all this data here and use the postDate to be the createdAt date" - it's not a 'normal'  operation. (Unless I've missed something).

**Comment by chapmandu** [comment]
Closed 9da9441
